core/citrus-api/src/main/java/com/consol/citrus/util/VariableExpressionIterator.java

```diff
@@ -0,0 +1,179 @@
+        }
+
+        if (StringUtils.hasLength(index)) {
+            segmentValue = getIndexedElement(name, index, segmentValue);
+        }
+
+        return new VariableSegment(name, index, segmentValue);
+    }
+
+    private Object getValueFromObjectField(String name, Object parentValue) {
+        Field field = ReflectionUtils.findField(parentValue.getClass(), name);
+        if (field == null) {
+            throw new CitrusRuntimeException(String.format("Failed to get variable - unknown field '%s' on type %s",
+                    name, parentValue.getClass().getName()));
+        }
+
+        ReflectionUtils.makeAccessible(field);
+        return ReflectionUtils.getField(field, parentValue);
+    }
+
+    private Object getValueFromContext(String name) {
+        Object matchedVariableValue = null;
+        if (testContext.getVariables().containsKey(name)) {
+            matchedVariableValue = testContext.getVariables().get(name);
+        } else {
+            throw new CitrusRuntimeException("Unknown variable '" + variableExpression + "'");
+        }
+        return matchedVariableValue;
+    }
+
+    private Object getIndexedElement(String name, String index, Object arrayValue) {
+        Object indexValue = null;
+        if (arrayValue.getClass().isArray()) {
+            indexValue = Array.get(arrayValue, Integer.valueOf(index));
+        } else {
+            throw new CitrusRuntimeException(
+                    String.format("Expected an instance of Array type. Cannot retrieve indexed property %s from %s ",
+                            name, arrayValue.getClass().getName()));
+        }
+        return indexValue;
+    }
+
+    public static Object getLastExpressionValue(String variableExpression, TestContext testContext) {
+        VariableSegment segment = null;
+        VariableExpressionIterator iterator = new VariableExpressionIterator(variableExpression, testContext);
+        while (iterator.hasNext()) {
+            segment = iterator.next();
+        }
+
+        return segment.getSegmentValue();
+    }
+
+    public static class VariableSegment {
+
+        /**
+         * The name of the variableExpression segment
+         */
+        private final String name;
+
+        /**
+         * An optional index if the VariableSegment represents an array
+         */
+        private final String index;
+
+        /**
+         * The evaluated value for this segment
+         */
+        private final Object segmentValue;
+
+        public VariableSegment(String name, String index, Object segmentValue) {
+            super();
+            this.name = name;
+            this.index = index;
+            this.segmentValue = segmentValue;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public String getIndex() {
+            return index;
+        }
+
+        public Object getSegmentValue() {
+            return segmentValue;
+        }
+    }
+
+}
```

Comment:
you could swap lines 178 and 179 - it's good pattern to have a newline at the end of the file.